### PR TITLE
Add JvmOverloads for Compass plugin

### PR DESCRIPTION
**Pull request by axti**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### Summary of changes
- Add JvmOverloads for Compass View constructor to simplify the code
- Move compass view size to dimensions.

### CODE REVIEW

**Comment by Chaoba** [comment]
IMO, we don't need to add `JvmOverloads` for plugin classes because we create the plugin in our SDK, users don't need to create a plugin instance.

File: plugin-compass/src/main/java/com/mapbox/maps/plugin/compass/CompassViewImpl.kt [hunk 1]
```diff
@@ -50,8 +36,8 @@ open class CompassViewImpl : CompassView, AppCompatImageView {
     )?.let { compassImage = it }
 
     // Layout params
-    val screenDensity = context.resources.displayMetrics.density
-    layoutParams = FrameLayout.LayoutParams((48 * screenDensity).toInt(), (48 * screenDensity).toInt())
+    val compassViewSize = resources.getDimensionPixelSize(R.dimen.compass_view_size)
+    layoutParams = FrameLayout.LayoutParams(compassViewSize, compassViewSize)
```
Comment (tobrun) [inline plugin-compass/src/main/java/com/mapbox/maps/plugin/compass/CompassViewImpl.kt:40]: :+1:

File: plugin-compass/src/main/java/com/mapbox/maps/plugin/compass/CompassViewPlugin.kt [hunk 2]
```diff
@@ -48,38 +49,39 @@ open class CompassViewPlugin(
...
-    )
-    fadeAnimator.addUpdateListener {
-      val value = it.animatedValue as Float
-      if (value < internalSettings.opacity) {
-        compassView.setCompassAlpha(value)
+      addUpdateListener {
+        val value = it.animatedValue as Float
+        if (value < internalSettings.opacity) {
+          compassView.setCompassAlpha(value)
+        }
```
Comment (tobrun) [inline plugin-compass/src/main/java/com/mapbox/maps/plugin/compass/CompassViewPlugin.kt:62]: :broom:

**Review by tobrun — APPROVED** [review]

**Comment by tobrun** [comment]
> IMO, we don't need to add JvmOverloads for plugin classes because we create the plugin in our SDK, users don't need to create a plugin instance.

Maybe not a pure requirement but as with all the changes made in this PR it's cleaner and also correct.

File: plugin-compass/src/main/java/com/mapbox/maps/plugin/compass/CompassViewImpl.kt (hunk 1 above)
Comment (axti) [reply plugin-compass/src/main/java/com/mapbox/maps/plugin/compass/CompassViewImpl.kt:40]: If some of customers would like to change compass size (for example for automotive maybe it is good to have it bigger) they can override this dimension and we don't need to update or modify our code.

Comment (alexander-kulikovskii) [reply plugin-compass/src/main/java/com/mapbox/maps/plugin/compass/CompassViewImpl.kt:40]: good point
Should we think about min size? If I set size to `0` I'll not see this element. Of course it's my fault as a developer, but maybe we should print some alert in console?

**Review by kiryldz — APPROVED** [review]
@axti is it a breaking change? for Java users?
Please update changelog entry as well

**Comment by axti** [comment]
> @axti is it a breaking change? for Java users? Please update changelog entry as well

There is no breaking changes, just cleaning and applying @JvmOverload

**Pull request merged**